From ff1478653fca993f473c36a37ab20614b71fb34c Mon Sep 17 00:00:00 2001
From: Josef Holzmayr <jester@theyoctojester.info>
Date: Wed, 10 Sep 2025 07:59:11 +0000
Subject: [PATCH] fix: disable bootdelay/keypress detection on Pi5-family

The Raspberry Pi 5 and Compute Module 5 seem to receive spurious
characters on the debug UART which interrupts the boot process.

Setting bootdelay to -2 prevents u-boot from stopping on the serial
console.

Signed-off-by: Josef Holzmayr <jester@theyoctojester.info>

Upstream-Status: Inappropriate [Mender specific]
---
 configs/rpi_arm64_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/rpi_arm64_defconfig b/configs/rpi_arm64_defconfig
index 2710613d072..3082988d60d 100644
--- a/configs/rpi_arm64_defconfig
+++ b/configs/rpi_arm64_defconfig
@@ -67,3 +67,4 @@ CONFIG_ENV_IS_IN_MMC=y
 CONFIG_ENV_OFFSET=0x800000
 CONFIG_ENV_OFFSET_REDUND=0x1000000
 CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
+CONFIG_BOOTDELAY=-2
-- 
2.34.1

